### ISSUE TITLE
ReferenceError: __nollup__external__util__default__ is not defined

### ISSUE THREAD

**Issue by charlag** [PepsRyuu/nollup#163]
For an empty file with just `import o from "ospec"` (which resolves to a file) nollup generated this:
```js
// ...
6:function (__c__, __r__, __d__, __e__) {
            

            var ex_default = __nollup__external__util__default__;
// ...
```
it's never used but it's not defined either

**Comment by PepsRyuu** [comment]
Reproducible example?

That should only happen if there's something like "export { default } from 'util'" happening somewhere.

**Comment by charlag** [comment]
I can share the branch, it's a converted commonjs module.
```sh
git clone git@github.com:tutao/tutanota.git
cd tutanota
npm ci
npm run testapi
```

**Comment by charlag** [comment]
Ah, sorry, I forgot that you need a special branch of ospec from here https://github.com/tutao/ospec/tree/dynamic-import
This is the code basically:
https://github.com/tutao/ospec/blob/dynamic-import/ospec.js#L43-47

I will try to narrow it down for a better example

**Comment by charlag** [comment]
Ah, sorry again for confusion and spam but I got confused myself.
The problem was that it was *not* the branch I mentioned and that's why it was `requite("util")` instead of `import("util")`. My code resolves it as `false`.

I can't manage to reproduce thiss in isolation so far...

**Comment by PepsRyuu** [comment]
Is there any update on this? I'm very confused. 😨

**Comment by charlag** [comment]
It should have happened when there's unresolved import but I couldn't reproduce it yet so I'll close for now

**Comment by charlag** [comment]
I've run into it again.
It's easy to reproduce:
- start with dynamic import
- import something external non-dynamically in the second file

like:

index.js:

```js
import("./util1")
```

util1.js:

```js

import fs from "fs"

fs.readFileSync("./index.js")
```

provided that "fs" is resolved as external (with plugin-node-resolve or own plugin which resolves to false) it will fail with `__nollup__external__fs__default__` because there's no import for it.

I've created a branch called `external-undefined` on https://github.com/tutao/nollup-bug/tree/external-undefined

**Comment by PepsRyuu** [comment]
Nice one! Will investigate later this evening. :)

**Comment by charlag** [comment]
I'll have a look myself too, maybe I'll manage to pin it

**Comment by charlag** [comment]
idk if it's a good solution but it's a workaround, maybe it'll help you later:

https://github.com/tutao/nollup/commit/c74e73b1600dd2b8207699d4d769b8643d5cc5e5

**Comment by PepsRyuu** [comment]
Copied the fix as suggested. Will be released in next version (```0.14.4```)

**Comment by charlag** [comment]
thanks a lot!
